spec/controllers/api/v1/printers_users_controller_spec.rb

```diff
@@ -0,0 +1,76 @@
+require "rails_helper"
+
+RSpec.describe Api::V1::PrintersUsersController, type: :controller do
+  let(:supervisor) { create(:user, :supervisor, :with_can_access_printers_permission)}
```

Comment:
I think `can_access_printers` isn't enough on its own as a permission.
Because looking at this, it seems it would allow someone with e.g `Stock fulfilment` role to modify a `Supervisor`'s favourite printers.

Those "lower" roles should only be allowed to create/edit their own printers. We probably need a `can_manage_printers` 

**Important**: before doing anything, write the tests that prove this is true

cc @steveyken